src/main/java/com/redhat/devtools/lsp4ij/features/codeBlockProvider/LSPCodeBlockProvider.java

```diff
@@ -78,50 +92,65 @@ public TextRange getCodeBlockRange(Editor editor, PsiFile file) {
+            );
+        }
+
+        // If those failed and we're seemingly anchored by a brace character, try to search unanchored
+        if ((codeBlockRange == null) && (openBraceChar != null) && (closeBraceChar != null)) {
+            codeBlockRange = getUsingSelectionRanges(
+                    file,
+                    editor,
+                    offset,
+                    null,
+                    -1,
+                    null,
+                    -1
+            );
+
+            if (codeBlockRange == null) {
+                codeBlockRange = getUsingFoldingRanges(
+                        file,
+                        document,
+                        offset,
+                        null,
+                        -1,
+                        null,
+                        -1
+                );
+            }
+        }
+
+        return codeBlockRange;
     }
 
     @Nullable
-    private TextRange getUsingSelectionRanges(@NotNull PsiFile file,
-                                              @NotNull Editor editor,
-                                              int offset,
-                                              @Nullable Character openBraceChar,
-                                              int openBraceOffset,
-                                              @Nullable Character closeBraceChar,
-                                              int closeBraceOffset) {
+    private static TextRange getUsingSelectionRanges(@NotNull PsiFile file,
+                                                     @NotNull Editor editor,
+                                                     int offset,
+                                                     @Nullable Character openBraceChar,
+                                                     int openBraceOffset,
+                                                     @Nullable Character closeBraceChar,
+                                                     int closeBraceOffset) {
         Document document = editor.getDocument();
-        List<SelectionRange> selectionRanges = LSPSelectionRangeSupport.getSelectionRanges(file, document, offset);
-        if (!ContainerUtil.isEmpty(selectionRanges)) {
-            // Convert the selection ranges into text ranges
-            Set<TextRange> textRanges = new LinkedHashSet<>(selectionRanges.size());
-            for (SelectionRange selectionRange : selectionRanges) {
-                textRanges.add(LSPSelectionRangeSupport.getTextRange(selectionRange, document));
-                for (SelectionRange parentSelectionRange = selectionRange.getParent();
-                     parentSelectionRange != null;
-                     parentSelectionRange = parentSelectionRange.getParent()) {
-                    textRanges.add(LSPSelectionRangeSupport.getTextRange(parentSelectionRange, document));
-                }
-            }
-
+        List<TextRange> selectionTextRanges = LSPSelectionRangeSupport.getSelectionTextRanges(file, editor, offset);
```

Comment:
This is just a result of additional encapsulation of how selection ranges are converted into text ranges.